core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/ngbd-modal-workflow-executions/ngbd-modal-workflow-executions.component.ts

```diff
@@ -266,6 +268,42 @@ export class NgbdModalWorkflowExecutionsComponent implements OnInit {
     }
   }
 
+  shortenName(name: string): string {
+    if (name.length <= 25) {
+      return name;
+    } else {
+      let words = name.slice(0, 25).split(" ");
+      return words.slice(0, -1).join(" ") + "...";
+    }
+  }
+
+  onHit(column: string, index: number): void {
+    if (this.showORhide[index]) {
+      this.ascSort(column);
+    } else {
+      this.dscSort(column);
+    }
+    this.showORhide[index] = !this.showORhide[index];
+  }
+
+  setAvatarColor(userName: string): string {
+    if (userName in this.avatarColors) {
+      return this.avatarColors[userName];
+    } else {
+      this.avatarColors[userName] = this.getRandomColor();
+      return this.avatarColors[userName];
+    }
+  }
+
+  getRandomColor(): string {
```

Comment:
can you please search how other components in Texera are using Random avatar color?